spec/features/management/account_spec.rb

```diff
@@ -29,4 +29,91 @@
     expect(user.reload.erase_reason).to eq "Deleted by manager: manager_user_#{Manager.last.user_id}"
   end
 
-end
\ No newline at end of file
+  scenario "Send reset password email to currently managed user session" do
+    user = create(:user, :level_three)
+    login_managed_user(user)
+    visit management_account_path
+
+    click_link 'Reset password'
+
+    expect(page).to have_content 'Edit user account: Reset password'
+
+    click_link 'Send reset password email'
+
+    expect(page).to have_content 'Email correctly sent.'
+
+    logout
+
+    sent_token = /.*reset_password_token=(.*)".*/.match(ActionMailer::Base.deliveries.last.body.to_s)[1]
+    visit edit_user_password_path(reset_password_token: sent_token)
+
+    fill_in 'user_password', with: 'new password'
+    fill_in 'user_password_confirmation', with: 'new password'
+    click_button 'Change my password'
+
+    expect(page).to have_content "Your password has been changed successfully."
+  end
+
+  scenario "Manager changes the password by hand (writen by them)" do
+    user = create(:user, :level_three)
+    login_managed_user(user)
+    visit management_account_path
+
+    click_link 'Reset password'
+
+    find(:css, "input[id$='user_password']").set("new_password")
+
+    click_button 'Save'
+
+    expect(page).to have_content 'Password reseted successfully'
+    expect(page).to have_link 'Print password'
+
+    logout
+
+    login_through_form_with_email_and_password(user.email, 'new_password')
+
+    expect(page).to have_content 'You have been signed in successfully.'
+  end
+
+  scenario "Manager generates random password", :js do
+    user = create(:user, :level_three)
+    user.update(email: nil)
+    login_managed_user(user)
+    visit management_account_path
+
+    click_link 'Reset password'
+    click_link 'Generate random password'
+
+    new_password = find_field('user_password').value
+
+    click_button 'Save'
+
+    expect(page).to have_content 'Password reseted successfully'
+    expect(page).to have_link 'Print password'
+
+    logout
+
+    login_through_form_with_email_and_password(user.username, new_password)
+
+    expect(page).to have_content 'You have been signed in successfully.'
+  end
+
+  scenario "The password is printed", :js do
+    user = create(:user, :level_three)
+    user.update(email: nil)
+    login_managed_user(user)
+    visit management_account_path
+
+    click_link 'Reset password'
+    click_link 'Generate random password'
+
+    new_password = find_field('user_password').value
+
+    click_button 'Save'
+
+    expect(page).to have_content 'Password reseted successfully'
+    expect(page).to have_css("a[href='javascript:window.print();']", text: 'Print password')
+    expect(page).to have_css("div.for-print-only", text: /Password:/, visible: false)
```

Comment:
Are we really checking here that the print functionallity works? Or just that the links are present?

Maybe it would be better to use the "manually set password" scenario to be able to correctly check that when printing we're showing the same password that the manager typed

Reply:
I changed that spec, and now the test sets a password "manualy" and then checks that the new password is in the page.
However, I didn't merge two specs together (the one that sets the password manually and this one), because in my opinion both of them check different things: one checks that the password has been correctly set and the user has no problems to log in after that; the other one checks that after saving the password, the admin has the option to print it (looking for the button) and the written password is present only for printing (I couldn't find any way to check the print dialog :confused: )

Reply:
Yes, I expressed myself poorly. I didn't meant to merge both scenarios (the `Manager changes the password by hand (writen by them)`  and `The password is printed`, but to use a manually written password on the `The password is printed` scenario to be able to check it in the printed view (as you correctly did 👏 )